staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_whitebox_test.go

```diff
@@ -258,39 +261,43 @@ func TestShouldDelegateList(t *testing.T) {
 	testCases[opts{ResourceVersion: "0", Continue: continueOnNegativeRV}] = true
 	testCases[opts{ResourceVersion: "0", Limit: 100, Continue: continueOnNegativeRV}] = true
 	testCases[opts{ResourceVersion: "0", ResourceVersionMatch: metav1.ResourceVersionMatchNotOlderThan}] = false
 	testCases[opts{ResourceVersion: "0", ResourceVersionMatch: metav1.ResourceVersionMatchNotOlderThan, Limit: 100}] = false
-	testCases[opts{ResourceVersion: "1"}] = false
-	testCases[opts{ResourceVersion: "1", Limit: 100}] = true
-	testCases[opts{ResourceVersion: "1", ResourceVersionMatch: metav1.ResourceVersionMatchExact}] = true
-	testCases[opts{ResourceVersion: "1", ResourceVersionMatch: metav1.ResourceVersionMatchExact, Limit: 100}] = true
-	testCases[opts{ResourceVersion: "1", ResourceVersionMatch: metav1.ResourceVersionMatchNotOlderThan}] = false
-	testCases[opts{ResourceVersion: "1", ResourceVersionMatch: metav1.ResourceVersionMatchNotOlderThan, Limit: 100}] = false
+	for _, rv := range []string{oldRV, consistantRV} {
+		testCases[opts{ResourceVersion: rv}] = false
+		testCases[opts{ResourceVersion: rv, Limit: 100}] = true
+		testCases[opts{ResourceVersion: rv, ResourceVersionMatch: metav1.ResourceVersionMatchExact}] = true
+		testCases[opts{ResourceVersion: rv, ResourceVersionMatch: metav1.ResourceVersionMatchExact, Limit: 100}] = true
+		testCases[opts{ResourceVersion: rv, ResourceVersionMatch: metav1.ResourceVersionMatchNotOlderThan}] = false
+		testCases[opts{ResourceVersion: rv, ResourceVersionMatch: metav1.ResourceVersionMatchNotOlderThan, Limit: 100}] = false
+	}
 
 	// Bypass for most requests doesn't depend on Recursive
 	for opts, expectBypass := range testCases {
 		opts.Recursive = true
 		testCases[opts] = expectBypass
 	}
 	// Continue is ignored on non recursive LIST
-	testCases[opts{ResourceVersion: "1", Continue: continueOnRev1}] = true
-	testCases[opts{ResourceVersion: "1", Continue: continueOnRev1, Limit: 100}] = true
-	testCases[opts{ResourceVersion: "1", Continue: continueOnNegativeRV}] = true
-	testCases[opts{ResourceVersion: "1", Continue: continueOnNegativeRV, Limit: 100}] = true
+	for _, rv := range []string{oldRV, consistantRV} {
+		testCases[opts{ResourceVersion: rv, Continue: continueOnOldRV}] = true
+		testCases[opts{ResourceVersion: rv, Continue: continueOnOldRV, Limit: 100}] = true
+		testCases[opts{ResourceVersion: rv, Continue: continueOnNegativeRV}] = true
+		testCases[opts{ResourceVersion: rv, Continue: continueOnNegativeRV, Limit: 100}] = true
+	}
 
-	for _, rv := range []string{"", "0", "1"} {
+	for _, rv := range []string{"", "0", oldRV, consistantRV} {
 		for _, match := range []metav1.ResourceVersionMatch{"", metav1.ResourceVersionMatchExact, metav1.ResourceVersionMatchNotOlderThan} {
-			for _, continueKey := range []string{"", continueOnRev1, continueOnNegativeRV} {
+			for _, continueKey := range []string{"", continueOnOldRV, continueOnNegativeRV} {
```

Comment:
nit: changes to this test seem orthogonal to this change - can we move them to a separate PR?